src/main/java/com/google/sps/servlets/PostComments.java

```diff
@@ -0,0 +1,110 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.Error;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Responsible for storing comments. */
+@SuppressWarnings("serial")
+@WebServlet("/post-comment")
+public class PostComments extends HttpServlet {
+  class CommentData {
+    Long routeId;
+    String commentText;
+
+    CommentData(Long routeId, String commentText) {
+      this.routeId = routeId;
+      this.commentText = commentText;
+    }
+
+    Long getRouteId() {
+      return routeId;
+    }
+
+    String getCommentText() {
+      return commentText;
+    }
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+
+    // Check if user is logged in.
+    if (!userService.isUserLoggedIn()) {
+      Error userError = new Error("You are not logged in!");
+      response.getWriter().println(gson.toJson(userError));
+      return;
+    }
+
+    // Read request body.
+    String requestBody =
+        request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+
+    // Convert it to Gson Object.
+    CommentData commentData = gson.fromJson(requestBody, CommentData.class);
+
+    // Get the input from the form.
+    String comment = commentData.getCommentText();
+    Long routeId = commentData.getRouteId();
+
+    Key routeKey = KeyFactory.createKey("Route", routeId);
+    Key userKey = KeyFactory.createKey("User", userService.getCurrentUser().getUserId());
+
+    try {
+      Entity userEntity = datastore.get(userKey);
+
+      // Store the comments as entities.
+      Entity commentEntity = new Entity("Comment", routeKey);
+      commentEntity.setProperty("text", comment);
+      commentEntity.setProperty("nickname", userEntity.getProperty("nickname"));
+
+      datastore.put(commentEntity);
+      Error errorStatus = new Error("Comment submitted!");
```

Comment:
This seems weird to create an error on success. It might be better to call the class Result or Status, and add a flag to the class whether it was successful or not.

Reply:
Done